### ISSUE TITLE
Backpropagation

### ISSUE THREAD

**Issue by SzilvasiPeter** [evoRegression/Regression#79]
- [x] Calculate loss function. Use Least square error: 1/(2*m) * SUM( (a-y)^2 )
- [x] Calculate the error function derivative respect to activation function: dE/da = 1/m * SUM( (a-y) )
- [x] Calculate the activation function derivatives: da/dz (dsigmoid, drelu, dtanh)
- [x] Calculate weighted sum respect to weights and bias: dz/dw = x; dz/da = 1

Gradients

- [x] Calculate cost gradient respect to weights: dE/dw = dE/da*da/dz*dz/dw
- [x] Calculate cost gradient respect to weights: dE/dw = dE/da*da/dz*dz/db

Update statement

- [x] Update weights: w = w - alpha*(dE/dw)
- [x] Update bias: b = b - alpha*(dE/db)
